diff -rupN originalLc/usr/share/lipstick-jolla-home-qt5/eventsview/EventsViewList.qml patchLc/usr/share/lipstick-jolla-home-qt5/eventsview/EventsViewList.qml
--- originalLc/usr/share/lipstick-jolla-home-qt5/eventsview/EventsViewList.qml	2018-06-21 06:26:58.475767207 +0100
+++ patchLc/usr/share/lipstick-jolla-home-qt5/eventsview/EventsViewList.qml	2018-06-21 06:27:34.655769389 +0100
@@ -175,7 +175,12 @@ SilicaFlickable {
             active: false
             eventsView: root.eventsView
         }
-
+    ConfigurationValue {
+        id: kmOn
+        key: "/desktop/lipstick-jolla-home/kidsMode/kmOn"
+        defaultValue: false
+        }
+        
         ConfigurationValue {
             id: eventsScreenWidgets
 
@@ -231,7 +236,8 @@ SilicaFlickable {
 
     Item {
         id: notificationsArea
-
+        
+        visible: !kmOn.value
         width: root.notificationWidth
         height: Math.max(systemUpdateList.contentHeight + notificationList.contentHeight + feedsList.height, root.height - y)
 
diff -rupN originalLc/usr/share/lipstick-jolla-home-qt5/eventsview/EventsView.qml patchLc/usr/share/lipstick-jolla-home-qt5/eventsview/EventsView.qml
--- originalLc/usr/share/lipstick-jolla-home-qt5/eventsview/EventsView.qml	2018-06-21 06:26:58.435767204 +0100
+++ patchLc/usr/share/lipstick-jolla-home-qt5/eventsview/EventsView.qml	2018-06-21 06:27:34.615769387 +0100
@@ -213,6 +213,11 @@ MouseArea {
             defaultValue: true
         }
 
+        ConfigurationValue {
+            id: kmOn
+            key: "/desktop/lipstick-jolla-home/kidsMode/kmOn"
+            defaultValue: false
+        }
         Loader {
             id: shortcutLoader
             active: shortcutsOverlay.active
@@ -352,6 +357,7 @@ MouseArea {
                                 rightMargin: Theme.paddingLarge
                             }
                             icon.source: "image://theme/icon-s-setting"
+                            visible: !kmOn.value
 
                             onClicked: {
                                 favoriteSettingsLoader.showEventsSettings()
diff -rupN originalLc/usr/share/lipstick-jolla-home-qt5/eventsview/FavoriteSettingsDelegate.qml patchLc/usr/share/lipstick-jolla-home-qt5/eventsview/FavoriteSettingsDelegate.qml
--- originalLc/usr/share/lipstick-jolla-home-qt5/eventsview/FavoriteSettingsDelegate.qml	2018-06-21 06:26:58.505767208 +0100
+++ patchLc/usr/share/lipstick-jolla-home-qt5/eventsview/FavoriteSettingsDelegate.qml	2018-06-21 06:27:34.705769392 +0100
@@ -9,6 +9,7 @@ import QtQuick 2.0
 import Sailfish.Silica 1.0
 import com.jolla.settings 1.0
 import org.nemomobile.lipstick 0.1
+import org.nemomobile.configuration 1.0
 import "../main"
 
 EditableGridDelegate {
@@ -46,7 +47,12 @@ EditableGridDelegate {
             value: item.down && item.enabled
         }
     }
-
+    ConfigurationValue {
+        id: kmOn
+        key: "/desktop/lipstick-jolla-home/kidsMode/kmOn"
+        defaultValue: false
+    }
+    
     Image {
         id: removeButton
         property bool pressed
@@ -65,7 +71,7 @@ EditableGridDelegate {
             horizontalCenterOffset: removeButtonOffset.width
             verticalCenterOffset: removeButtonOffset.height
         }
-        enabled: Lipstick.compositor.eventsLayer.shortcutsHousekeeping
+        enabled: Lipstick.compositor.eventsLayer.shortcutsHousekeeping && !kmOn.value
         opacity: enabled ? 1.0 : 0
         Behavior on opacity { FadeAnimation { } }
     }
diff -rupN originalLc/usr/share/lipstick-jolla-home-qt5/eventsview/QuickActionsLoader.qml patchLc/usr/share/lipstick-jolla-home-qt5/eventsview/QuickActionsLoader.qml
--- originalLc/usr/share/lipstick-jolla-home-qt5/eventsview/QuickActionsLoader.qml	2018-06-21 06:26:58.545767211 +0100
+++ patchLc/usr/share/lipstick-jolla-home-qt5/eventsview/QuickActionsLoader.qml	2018-06-21 06:27:34.745769394 +0100
@@ -10,6 +10,7 @@ import Sailfish.Silica 1.0
 import org.nemomobile.lipstick 0.1
 import com.jolla.settings 1.0
 import com.jolla.settings.system 1.0
+import org.nemomobile.configuration 1.0
 import "../main"
 
 Loader {
@@ -25,7 +26,12 @@ Loader {
 
     asynchronous: true
     visible: status === Loader.Ready
-
+    ConfigurationValue {
+        id: kmOn
+        key: "/desktop/lipstick-jolla-home/kidsMode/kmOn"
+        defaultValue: false
+    }
+    
     Rectangle {
         anchors {
             fill: parent
@@ -148,7 +154,7 @@ Loader {
                             horizontalCenter: parent.right
                             horizontalCenterOffset: Theme.iconSizeMedium/2-iconOffset
                         }
-                        enabled: wrapper.editMode
+                        enabled: wrapper.editMode && !kmOn.value
                         opacity: enabled ? 1.0 : 0
                         icon.source: "image://theme/icon-m-clear"
                         width: icon.width
diff -rupN originalLc/usr/share/lipstick-jolla-home-qt5/launcher/LauncherGrid.qml patchLc/usr/share/lipstick-jolla-home-qt5/launcher/LauncherGrid.qml
--- originalLc/usr/share/lipstick-jolla-home-qt5/launcher/LauncherGrid.qml	2018-06-21 06:28:08.815771450 +0100
+++ patchLc/usr/share/lipstick-jolla-home-qt5/launcher/LauncherGrid.qml	2018-06-21 20:16:22.346113755 +0100
@@ -31,6 +31,11 @@ SilicaGridView {
         property real zoomValue: 1.0
     }
 
+    ConfigurationGroup {
+                id: kidsModeSettings
+                path: "/desktop/lipstick-jolla-home/kidsMode"
+                property bool kmOn: false
+    }
     property bool isPortrait: launcherPager.height > launcherPager.width
 
     add: Transition {
@@ -264,7 +269,7 @@ SilicaGridView {
         }
 
         onPressAndHold: {
-            if (Lipstick.compositor.launcherLayer.active) {
+            if (Lipstick.compositor.launcherLayer.active && !kmOn.value) {
                 setEditMode(true)
             }
         }
@@ -371,7 +376,7 @@ SilicaGridView {
                 anchors.horizontalCenter: parent.horizontalCenter
                 enabled: launcherEditMode && policy.value
                 opacity: enabled ? 1.0 : 0.0
-                visible: launcherEditMode && !isFolder && AppControl.isUninstallable(object.filePath) && !object.isUpdating
+                visible: launcherEditMode && !isFolder && AppControl.isUninstallable(object.filePath) && !object.isUpdating  && !kidsModeSettings.kmOn
                 onClicked: removeApplication(object.filePath, object.title)
             }
         }
diff -rupN originalLc/usr/share/lipstick-jolla-home-qt5/launcher/Launcher.qml patchLc/usr/share/lipstick-jolla-home-qt5/launcher/Launcher.qml
--- originalLc/usr/share/lipstick-jolla-home-qt5/launcher/Launcher.qml	2018-06-21 06:28:08.815771450 +0100
+++ patchLc/usr/share/lipstick-jolla-home-qt5/launcher/Launcher.qml	2018-06-21 20:12:27.044828396 +0100
@@ -24,7 +24,32 @@ SilicaListView {
         property bool useScroll: true
         onFreeScrollChanged: launcher.manageDummyPages()
     }    
-
+    
+ConfigurationValue {
+        id: kmOn
+        key: "/desktop/lipstick-jolla-home/kidsMode/kmOn"
+        defaultValue: false
+    }
+        
+    ConfigurationValue {
+        id: lockScreenShortcuts
+
+        key: "/desktop/lipstick-jolla-home/kidsMode/"+currentUser.value+"/appsList"
+        defaultValue: []
+    }
+    ConfigurationValue {
+        id: currentUser
+
+        key: "/desktop/lipstick-jolla-home/kidsMode/currentUser"
+        defaultValue: 0
+    }
+            
+    ConfigurationGroup {
+        id: kmTrigger
+path: "/desktop/lipstick-jolla-home/kidsMode"
+        property bool triggerLoad: false
+        onValueChanged:if(key == "triggerLoad") lFolder.load() 
+    }
     property bool launcherActive: Lipstick.compositor.launcherLayer.active
     onLauncherActiveChanged: if (!launcherActive) { resetPosition(400) }
 
@@ -128,7 +153,7 @@ SilicaListView {
         }
 
         onPressAndHold: {
-            if (Lipstick.compositor.launcherLayer.active &&
+            if (Lipstick.compositor.launcherLayer.active && !kmOn.value &&
                 Math.abs(mouse.x - pressX) < Theme.startDragDistance &&
                 Math.abs(mouse.y - pressY) < Theme.startDragDistance) {
                 launcher.setEditMode(true)
@@ -145,11 +170,14 @@ SilicaListView {
             gridManager.onStopScrolling: launcherPager.stopScrolling()
 
             model: LauncherFolderModel {
+                id: lFolder
                 iconDirectories: Theme.launcherIconDirectories
-                blacklistedApplications: {
+                blacklistedApplications: kmOn.value? getBlackList(lockScreenShortcuts.value):getBlackList([ ])
+
+                function getBlackList(appList) {
                     // Currently desktop-file path is good app grid item
                     // identifier. However, this is a subject to change in future.
-                    var blacklist = []
+                    var blacklist = appList
                     var path = "/usr/share/applications"
 
                     if (!AccessPolicy.cameraEnabled) {
diff -rupN originalLc/usr/share/lipstick-jolla-home-qt5/notifications/NotificationPreview.qml patchLc/usr/share/lipstick-jolla-home-qt5/notifications/NotificationPreview.qml
--- originalLc/usr/share/lipstick-jolla-home-qt5/notifications/NotificationPreview.qml	2018-06-21 06:26:58.635767216 +0100
+++ patchLc/usr/share/lipstick-jolla-home-qt5/notifications/NotificationPreview.qml	2018-06-21 06:27:34.865769402 +0100
@@ -12,6 +12,7 @@ import com.jolla.lipstick 0.1
 import org.nemomobile.lipstick 0.1
 import org.nemomobile.thumbnailer 1.0
 import org.nemomobile.devicelock 1.0
+import org.nemomobile.configuration 1.0
 import "../systemwindow"
 
 SystemWindow {
@@ -63,6 +64,11 @@ SystemWindow {
         property: "iconUrl"
         value: notificationWindow.iconUrl
     }
+    ConfigurationValue {
+        id: kmOn
+        key: "/desktop/lipstick-jolla-home/kidsMode/kmOn"
+        defaultValue: false
+    }
 
     function firstLine(str) {
         var i = str.indexOf("\n")
@@ -137,7 +143,7 @@ SystemWindow {
         }
 
         onClicked: {
-            if (notification) {
+            if (notification && !kmOn.value) {
                 notificationWindow._invoked = true
                 notification.actionInvoked("default")
 
diff -rupN originalLc/usr/share/lipstick-jolla-home-qt5/switcher/Switcher.qml patchLc/usr/share/lipstick-jolla-home-qt5/switcher/Switcher.qml
--- originalLc/usr/share/lipstick-jolla-home-qt5/switcher/Switcher.qml	2018-06-21 06:26:58.675767219 +0100
+++ patchLc/usr/share/lipstick-jolla-home-qt5/switcher/Switcher.qml	2018-06-21 06:27:34.905769404 +0100
@@ -12,6 +12,7 @@ import com.jolla.lipstick 0.1
 import Sailfish.Silica 1.0
 import Sailfish.Silica.private 1.0
 import Sailfish.Lipstick 1.0
+import org.nemomobile.configuration 1.0
 import "../compositor"
 import "../main"
 
@@ -177,11 +178,143 @@ SilicaFlickable {
             repeater.itemAt(lastIndex).close()
         }
     }
+    Timer {
+        id: closeNonKmTimer
+        property int lastIndex
+
+        interval: 100
+        repeat: true
+        triggeredOnStart: true
+
+        onTriggered: {
+            if (repeater.count === 0 || lastIndex === 0) {
+                running = false
+                return
+            }
 
+            lastIndex = lastIndex - 1
+            if (lastIndex > repeater.count - 1) {
+                lastIndex = repeater.count - 1
+            }
+            var window = Lipstick.compositor.windowForId(repeater.itemAt(lastIndex).windowId)
+            if(window == null) repeater.itemAt(lastIndex).close()
+            else if(appTitles.value.indexOf(window.title) != -1) repeater.itemAt(lastIndex).close()
+        }
+    }
+    property var switcherModelBackup: []
+    
+    function hideApps() {     
+        switcherRoot.minimizeLaunchingWindows()
+        switcherRoot.closePending = true 
+        if(kmOn.value) {
+            var i = 0       
+            var items =[]   
+            
+            while(i < repeater.count) {   
+                var item = repeater.itemAt(i)
+                var window = Lipstick.compositor.windowForId(item.windowId)
+                if(window == null) {
+                    i = i + 1
+                    continue
+                }
+              if(appTitles.value.indexOf(window.title) != -1)   {
+                    if(item.launcherItem == null) {
+                       // item.minimized = true
+                      //  item.close()
+                        i = i + 1
+                        continue
+                    }
+                   
+                    items.push(item.launcherItem)
+                    items.push(switcherModel.windowId(i))
+                    items.push(item.coverId)
+                    items.push(item.windowId)
+                    items.push(item.coverHint)
+                    items.push(item.processId)
+                    items.push(item.windowMapPending)
+                    items.push(item.showingWid)
+                    items.push(item.running)
+                    items.push(item.pending)
+                    items.push(item.minimized)
+                    items.push(item._persistCloseApplicationEnabled)
+                    switcherModelBackup.push(items)
+                    switcherModel.remove(i)
+                    items =[]          
+                    
+                }
+                else {
+                        i = i + 1
+                }
+            }
+                switcherRoot.closeAll()  
+        }
+        else {
+            for (var i = 0; i <switcherModelBackup.length; ++i) {
+                var j =repeater.count                
+                var item =switcherModelBackup[i][0]
+                switcherModel.append(item)
+                var repItem = repeater.itemAt(j)
+                repItem.coverId = switcherModelBackup[i][2]
+                repItem.windowId = switcherModelBackup[i][3]
+                repItem.coverHint = switcherModelBackup[i][4]
+                repItem.processId = switcherModelBackup[i][5]
+                repItem.windowMapPending = switcherModelBackup[i][6]
+                repItem.showingWid = switcherModelBackup[i][7]
+                repItem.running = switcherModelBackup[i][8]
+                repItem.pending = switcherModelBackup[i][9]
+                repItem.minimized = switcherModelBackup[i][10]
+                repItem._persistCloseApplicationEnabled = switcherModelBackup[i][11]
+            }
+            switcherModelBackup = []
+        }
+       //switcherRoot.closePending = false
+    }   
+
+    ConfigurationGroup {
+        id: kmTrigger
+path: "/desktop/lipstick-jolla-home/kidsMode"
+        property bool triggerClose: false
+        property bool closeAllApps: true
+        onValueChanged: if(key == "triggerClose") closeAllApps ? switcherRoot.closeAll() :switcherRoot.hideApps()
+            
+    }  
+    
+    ConfigurationValue {
+        id: kmOn
+        key: "/desktop/lipstick-jolla-home/kidsMode/kmOn"
+        defaultValue: false
+    }
+    
+    ConfigurationValue {
+        id: currentUser
+        key: "/desktop/lipstick-jolla-home/kidsMode/currentUser"
+        defaultValue: 0
+    }
+    
+    ConfigurationValue {
+        id: lockScreenShortcuts
+
+        key: "/desktop/lipstick-jolla-home/kidsMode/"+currentUser.value+"/appsList"
+        defaultValue: []
+    }
+    
+    ConfigurationValue {
+        id: appTitles
+
+        key: "/desktop/lipstick-jolla-home/kidsMode/"+currentUser.value+"/appTitles"
+        defaultValue: []
+    }
+    
     function closeAll() {
+        if(kmTrigger.closeAllApps) {
         closeAllTimer.stop()
         closeAllTimer.lastIndex = repeater.count
-        closeAllTimer.start()
+        closeAllTimer.start()}
+        else {
+         closeNonKmTimer.stop()
+        closeNonKmTimer.lastIndex = repeater.count
+        closeNonKmTimer.start()}
+        
     }
 
     function isAndroidApplication(launcherItem) {
