diff -rupN originalLc/usr/share/lipstick-jolla-home-qt5/eventsview/EventsViewList.qml patchLc/usr/share/lipstick-jolla-home-qt5/eventsview/EventsViewList.qml
--- originalLc/usr/share/lipstick-jolla-home-qt5/eventsview/EventsViewList.qml	2018-11-20 16:00:56.448153112 +0000
+++ patchLc/usr/share/lipstick-jolla-home-qt5/eventsview/EventsViewList.qml	2018-11-20 16:02:34.808948074 +0000
@@ -27,7 +27,7 @@ SilicaFlickable {
     property real defaultNotificationAreaY: headerColumn.y + headerColumn.height + Theme.paddingMedium
     property bool menuOpen: pullDownMenu != null && pullDownMenu.active
 
-    property bool _housekeepingAllowed: (notificationListModel.populated && notificationList.hasRemovableNotifications)
+    property bool _housekeepingAllowed: kmOn.value ? false :(notificationListModel.populated && notificationList.hasRemovableNotifications)
                         || feedsList.hasRemovableNotifications
 
     contentHeight: Math.ceil(Math.max(footerSpacer.y + footerSpacer.height, noNotificationsLabel.y + noNotificationsLabel.height))
@@ -171,6 +171,23 @@ SilicaFlickable {
             eventsView: root
         }
 
+    ConfigurationValue {
+        id: kmOn
+        key: "/desktop/lipstick-jolla-home/kidsMode/kmOn"
+        defaultValue: false
+        }
+        ConfigurationValue {
+        id: showEvNotifications
+
+        key: "/desktop/lipstick-jolla-home/kidsMode/"+currentUser.value+"/showEvNotifications"
+        defaultValue: false
+    }
+    ConfigurationValue {
+        id: currentUser
+
+        key: "/desktop/lipstick-jolla-home/kidsMode/currentUser"
+        defaultValue: 0
+        }
         ConfigurationValue {
             id: eventsScreenWidgets
 
@@ -226,7 +243,8 @@ SilicaFlickable {
 
     Item {
         id: notificationsArea
-
+        
+        visible: kmOn.value ? showEvNotifications.value: true
         width: parent.width
         height: Math.max(systemUpdateList.contentHeight + notificationList.contentHeight + feedsList.height, root.height - y)
 
diff -rupN originalLc/usr/share/lipstick-jolla-home-qt5/launcher/LauncherGrid.qml patchLc/usr/share/lipstick-jolla-home-qt5/launcher/LauncherGrid.qml
--- originalLc/usr/share/lipstick-jolla-home-qt5/launcher/LauncherGrid.qml	2018-11-21 06:21:40.675382754 +0000
+++ patchLc/usr/share/lipstick-jolla-home-qt5/launcher/LauncherGrid.qml	2018-11-21 06:32:40.862126657 +0000
@@ -30,7 +30,11 @@ IconGridViewBase {
         property bool zoomFonts: false
         property real zoomValue: 1.0
     }
-
+    ConfigurationGroup {
+                id: kidsModeSettings
+                path: "/desktop/lipstick-jolla-home/kidsMode"
+                property bool kmOn: false
+    }
     property bool isPortrait: launcherPager.height > launcherPager.width
 
     add: Transition {
@@ -256,7 +260,7 @@ IconGridViewBase {
         }
 
         onPressAndHold: {
-            if (Lipstick.compositor.launcherLayer.active) {
+            if (Lipstick.compositor.launcherLayer.active && !kidsModeSettings.kmOn) {
                 setEditMode(true)
             }
         }
@@ -367,6 +371,7 @@ IconGridViewBase {
                             && !isFolder
                             && AppControl.isUninstallable(object.filePath)
                             && !object.isUpdating
+                            && !kidsModeSettings.kmOn
                             && object.readValue("X-apkd-apkfile").indexOf("/vendor/app/") != 0
                             && object.readValue("X-apkd-apkfile").indexOf("/home/.android/vendor/app/") != 0
                 onClicked: removeApplication(object.filePath, object.title)
diff -rupN originalLc/usr/share/lipstick-jolla-home-qt5/launcher/Launcher.qml patchLc/usr/share/lipstick-jolla-home-qt5/launcher/Launcher.qml
--- originalLc/usr/share/lipstick-jolla-home-qt5/launcher/Launcher.qml	2018-11-21 06:21:40.795377085 +0000
+++ patchLc/usr/share/lipstick-jolla-home-qt5/launcher/Launcher.qml	2018-11-21 06:37:57.972553887 +0000
@@ -24,7 +24,32 @@ SilicaListView {
         property bool useScroll: true
         onFreeScrollChanged: launcher.manageDummyPages()
     }    
-
+    
+ConfigurationValue {
+        id: kmOn
+        key: "/desktop/lipstick-jolla-home/kidsMode/kmOn"
+        defaultValue: false
+    }
+        
+    ConfigurationValue {
+        id: lockScreenShortcuts
+
+        key: "/desktop/lipstick-jolla-home/kidsMode/"+currentUser.value+"/appsList"
+        defaultValue: []
+    }
+    ConfigurationValue {
+        id: currentUser
+
+        key: "/desktop/lipstick-jolla-home/kidsMode/currentUser"
+        defaultValue: 0
+    }
+            
+    ConfigurationGroup {
+        id: kmTrigger
+path: "/desktop/lipstick-jolla-home/kidsMode"
+        property bool triggerLoad: false
+        onValueChanged:if(key == "triggerLoad") lFolder.load() 
+    }
     property bool launcherActive: Lipstick.compositor.launcherLayer.active
     onLauncherActiveChanged: if (!launcherActive) { resetPosition(400) }
 
@@ -128,7 +153,7 @@ SilicaListView {
         }
 
         onPressAndHold: {
-            if (Lipstick.compositor.launcherLayer.active &&
+            if (Lipstick.compositor.launcherLayer.active && !kmOn.value &&
                 Math.abs(mouse.x - pressX) < Theme.startDragDistance &&
                 Math.abs(mouse.y - pressY) < Theme.startDragDistance) {
                 launcher.setEditMode(true)
@@ -145,11 +170,14 @@ SilicaListView {
             gridManager.onStopScrolling: launcherPager.stopScrolling()
 
             model: LauncherFolderModel {
+                id: lFolder
                 iconDirectories: Theme.launcherIconDirectories
-                blacklistedApplications: {
+                blacklistedApplications:kmOn.value? getBlackList(lockScreenShortcuts.value):getBlackList([ ])
+
+                function getBlackList(appList) {
                     // Currently desktop-file path is good app grid item
                     // identifier. However, this is a subject to change in future.
-                    var blacklist = []
+                    var blacklist = appList
                     var path = "/usr/share/applications"
 
                     if (!AccessPolicy.cameraEnabled) {
diff -rupN originalLc/usr/share/lipstick-jolla-home-qt5/notifications/NotificationItem.qml patchLc/usr/share/lipstick-jolla-home-qt5/notifications/NotificationItem.qml
--- originalLc/usr/share/lipstick-jolla-home-qt5/notifications/NotificationItem.qml	2018-11-20 16:01:25.833453100 +0000
+++ patchLc/usr/share/lipstick-jolla-home-qt5/notifications/NotificationItem.qml	2018-11-20 16:02:47.632533069 +0000
@@ -11,6 +11,7 @@ import Sailfish.Lipstick 1.0
 import Nemo.DBus 2.0
 import com.jolla.lipstick 0.1
 import org.nemomobile.lipstick 0.1
+import org.nemomobile.configuration 1.0
 
 Item {
     id: root
@@ -40,7 +41,12 @@ Item {
     property int _timestampCounter
     property int _expansionThreshold: 3
     property int _expansionMaximum: 15
-
+    
+    ConfigurationValue {
+        id: kmOn
+        key: "/desktop/lipstick-jolla-home/kidsMode/kmOn"
+        defaultValue: false
+    }
     function _invokeAppAction() {
         // Invoke the 'app' action directly
         var remoteActions = root._notificationGroup.remoteActions
@@ -92,7 +98,7 @@ Item {
         }
 
         onTriggered: {
-            if (!root._notificationGroup || !root._notificationGroup.members.length) {
+            if (!root._notificationGroup || !root._notificationGroup.members.length || kmOn.value) {
                 return
             }
             root._invokeAppAction()
@@ -174,7 +180,7 @@ Item {
                     }
 
                     onTriggered: {
-                        notification.actionInvoked("default")
+                        if(!kmOn.value)   notification.actionInvoked("default")
                     }
                 }
             }
@@ -226,7 +232,7 @@ Item {
                 color: expansionToggle.highlighted ? Theme.secondaryHighlightColor : Theme.secondaryColor
             }
 
-            onClicked: root._invokeAppAction()
+            onClicked: if(!kmOn.value) root._invokeAppAction()
         }
     }
 }
diff -rupN originalLc/usr/share/lipstick-jolla-home-qt5/notifications/NotificationPreview.qml patchLc/usr/share/lipstick-jolla-home-qt5/notifications/NotificationPreview.qml
--- originalLc/usr/share/lipstick-jolla-home-qt5/notifications/NotificationPreview.qml	2018-11-20 16:01:25.943398100 +0000
+++ patchLc/usr/share/lipstick-jolla-home-qt5/notifications/NotificationPreview.qml	2018-11-20 16:02:47.662518069 +0000
@@ -13,13 +13,14 @@ import com.jolla.lipstick 0.1
 import org.nemomobile.lipstick 0.1
 import org.nemomobile.thumbnailer 1.0
 import org.nemomobile.devicelock 1.0
+import org.nemomobile.configuration 1.0
 import "../systemwindow"
 
 SystemWindow {
     id: notificationWindow
 
     property QtObject notification: notificationPreviewPresenter.notification
-    property bool showNotification: notification != null && (notification.previewBody || notification.previewSummary)
+    property bool showNotification: notification != null && (notification.previewBody || notification.previewSummary) &&(kmOn.value? showPopUps.value:true)
     property string summaryText: showNotification ? notification.previewSummary : ''
     property string bodyText: showNotification ? notification.previewBody : ''
     property bool popupPresentation: state == "showPopup" || state == "hidePopup"
@@ -64,6 +65,23 @@ SystemWindow {
         property: "iconUrl"
         value: notificationWindow.iconUrl
     }
+    ConfigurationValue {
+        id: kmOn
+        key: "/desktop/lipstick-jolla-home/kidsMode/kmOn"
+        defaultValue: false
+    }
+    ConfigurationValue {
+        id: showPopUps
+
+        key: "/desktop/lipstick-jolla-home/kidsMode/"+currentUser.value+"/showPopUps"
+        defaultValue: true
+    }
+    ConfigurationValue {
+        id: currentUser
+
+        key: "/desktop/lipstick-jolla-home/kidsMode/currentUser"
+        defaultValue: 0
+    }
 
     function firstLine(str) {
         var i = str.indexOf("\n")
@@ -127,7 +145,7 @@ SystemWindow {
         }
 
         onClicked: {
-            if (notification) {
+            if (notification && !kmOn.value) {
                 notificationWindow._invoked = true
                 notification.actionInvoked("default")
 
diff -rupN originalLc/usr/share/lipstick-jolla-home-qt5/switcher/Switcher.qml patchLc/usr/share/lipstick-jolla-home-qt5/switcher/Switcher.qml
--- originalLc/usr/share/lipstick-jolla-home-qt5/switcher/Switcher.qml	2018-11-20 16:01:40.576078095 +0000
+++ patchLc/usr/share/lipstick-jolla-home-qt5/switcher/Switcher.qml	2018-11-20 16:03:01.635528064 +0000
@@ -12,6 +12,7 @@ import com.jolla.lipstick 0.1
 import Sailfish.Silica 1.0
 import Sailfish.Silica.private 1.0
 import Sailfish.Lipstick 1.0
+import org.nemomobile.configuration 1.0
 import "../compositor"
 import "../main"
 
@@ -181,11 +182,142 @@ SilicaFlickable {
             repeater.itemAt(lastIndex).close()
         }
     }
+    Timer {
+        id: closeNonKmTimer
+        property int lastIndex
+
+        interval: 100
+        repeat: true
+        triggeredOnStart: true
+
+        onTriggered: {
+            if (repeater.count === 0 || lastIndex === 0) {
+                running = false
+                return
+            }
+
+            lastIndex = lastIndex - 1
+            if (lastIndex > repeater.count - 1) {
+                lastIndex = repeater.count - 1
+            }
+            var window = Lipstick.compositor.windowForId(repeater.itemAt(lastIndex).windowId)
+            if(window == null) repeater.itemAt(lastIndex).close()
+            else if(appTitles.value.indexOf(window.title) != -1) repeater.itemAt(lastIndex).close()
+        }
+    }
+    property var switcherModelBackup: []
+    
+    function hideApps() {     
+        switcherRoot.minimizeLaunchingWindows()
+        switcherRoot.closePending = true 
+        if(kmOn.value) {
+            var i = 0       
+            var items =[]   
+            
+            while(i < repeater.count) {   
+                var item = repeater.itemAt(i)
+                var window = Lipstick.compositor.windowForId(item.windowId)
+                if(window == null) {
+                    i = i + 1
+                    continue
+                }
+              if(appTitles.value.indexOf(window.title) != -1)   {
+                    if(item.launcherItem == null) {
+                       // item.minimized = true
+                      //  item.close()
+                        i = i + 1
+                        continue
+                    }
+                   
+                    items.push(item.launcherItem)
+                    items.push(switcherModel.windowId(i))
+                    items.push(item.coverId)
+                    items.push(item.windowId)
+                    items.push(item.coverHint)
+                    items.push(item.processId)
+                    items.push(item.windowMapPending)
+                    items.push(item.showingWid)
+                    items.push(item.running)
+                    items.push(item.pending)
+                    items.push(item.minimized)
+
+                    switcherModelBackup.push(items)
+                    switcherModel.remove(i)
+                    items =[]          
+                    
+                }
+                else {
+                        i = i + 1
+                }
+            }
+                switcherRoot.closeAll()  
+        }
+        else {
+            for (var i = 0; i <switcherModelBackup.length; ++i) {
+                var j =repeater.count                
+                var item =switcherModelBackup[i][0]
+                switcherModel.append(item)
+                var repItem = repeater.itemAt(j)
+                repItem.coverId = switcherModelBackup[i][2]
+                repItem.windowId = switcherModelBackup[i][3]
+                repItem.coverHint = switcherModelBackup[i][4]
+                repItem.processId = switcherModelBackup[i][5]
+                repItem.windowMapPending = switcherModelBackup[i][6]
+                repItem.showingWid = switcherModelBackup[i][7]
+                repItem.running = switcherModelBackup[i][8]
+                repItem.pending = switcherModelBackup[i][9]
+                repItem.minimized = switcherModelBackup[i][10]
+           //     repItem._persistCloseApplicationEnabled = switcherModelBackup[i][11]
+            }
+            switcherModelBackup = []
+        }
+       //switcherRoot.closePending = false
+    }   
+
+    ConfigurationGroup {
+        id: kmTrigger
+path: "/desktop/lipstick-jolla-home/kidsMode"
+        property bool triggerClose: false
+        property bool closeAllApps: true
+        onValueChanged: if(key == "triggerClose") closeAllApps ? switcherRoot.closeAll() :switcherRoot.hideApps()
+            
+    }  
+    
+    ConfigurationValue {
+        id: kmOn
+        key: "/desktop/lipstick-jolla-home/kidsMode/kmOn"
+        defaultValue: false
+    }
+    
+    ConfigurationValue {
+        id: currentUser
+        key: "/desktop/lipstick-jolla-home/kidsMode/currentUser"
+        defaultValue: 0
+    }
+    
+    ConfigurationValue {
+        id: lockScreenShortcuts
+
+        key: "/desktop/lipstick-jolla-home/kidsMode/"+currentUser.value+"/appsList"
+        defaultValue: []
+    }
+    
+    ConfigurationValue {
+        id: appTitles
+
+        key: "/desktop/lipstick-jolla-home/kidsMode/"+currentUser.value+"/appTitles"
+        defaultValue: []
+    }
 
     function closeAll() {
+        if(kmTrigger.closeAllApps) {
         closeAllTimer.stop()
         closeAllTimer.lastIndex = repeater.count
-        closeAllTimer.start()
+        closeAllTimer.start()}
+        else {
+         closeNonKmTimer.stop()
+        closeNonKmTimer.lastIndex = repeater.count
+        closeNonKmTimer.start()}
     }
 
     function isAndroidApplication(launcherItem) {
diff -rupN originalLc/usr/share/lipstick-jolla-home-qt5/topmenu/AmbienceSelector.qml patchLc/usr/share/lipstick-jolla-home-qt5/topmenu/AmbienceSelector.qml
--- originalLc/usr/share/lipstick-jolla-home-qt5/topmenu/AmbienceSelector.qml	2018-11-20 16:01:53.419653090 +0000
+++ patchLc/usr/share/lipstick-jolla-home-qt5/topmenu/AmbienceSelector.qml	2018-11-20 16:03:14.029328059 +0000
@@ -44,7 +44,12 @@ Item {
         key: "/desktop/lipstick-jolla-home/topmenu_ambiences_enabled"
         defaultValue: true
     }
-
+    
+    ConfigurationValue {
+        id: kmOn
+        key: "/desktop/lipstick-jolla-home/kidsMode/kmOn"
+        defaultValue: false
+    }
     DBusInterface {
         id: settingsApp
 
@@ -95,7 +100,7 @@ Item {
 
         delegate: ListItem {
             id: thumbnailBackground
-
+            visible: kmOn.value? model.favorite : true
             readonly property bool dimmed: ambienceList.__silica_contextmenu_instance
                                            && ambienceList.__silica_contextmenu_instance.active
                                            && ambienceList.__silica_contextmenu_instance.parent != thumbnailBackground
@@ -103,7 +108,7 @@ Item {
             width: root.itemSize
             contentHeight: width
             highlightedColor: Theme.rgba(highlightBackgroundColor || Theme.highlightBackgroundColor, Theme.highlightBackgroundOpacity)
-            openMenuOnPressAndHold: Desktop.deviceLockState === DeviceLock.Unlocked
+            openMenuOnPressAndHold: Desktop.deviceLockState === DeviceLock.Unlocked && !kmOn.value
 
             onClicked: {
                 ambienceModel.makeCurrent(index)
diff -rupN originalLc/usr/share/lipstick-jolla-home-qt5/topmenu/TopMenu.qml patchLc/usr/share/lipstick-jolla-home-qt5/topmenu/TopMenu.qml
--- originalLc/usr/share/lipstick-jolla-home-qt5/topmenu/TopMenu.qml	2018-11-20 16:01:53.459633090 +0000
+++ patchLc/usr/share/lipstick-jolla-home-qt5/topmenu/TopMenu.qml	2018-11-20 16:03:14.059313059 +0000
@@ -216,7 +216,12 @@ SilicaFlickable {
             // Don't divide by zero.
             innerMargin: toggleColumns > 0 ? (width / toggleColumns) - Theme.itemSizeSmall : 0
         }
-
+        
+        ConfigurationValue {
+            id: kmOn
+            key: "/desktop/lipstick-jolla-home/kidsMode/kmOn"
+            defaultValue: false
+        }
         Loader {
             id: shortcutsLoader
             width: parent.width
@@ -281,6 +286,7 @@ SilicaFlickable {
                         IconButton {
                             id: settingsButton
 
+                            visible: !kmOn.value
                             function openMenu() {
                                 if (!topMenu.contextMenu) {
                                     topMenu.contextMenu = contextMenuComponent.createObject(topMenu)
