diff -rupN original/usr/share/lipstick-jolla-home-qt5/eventsview/EventsViewList.qml patch/usr/share/lipstick-jolla-home-qt5/eventsview/EventsViewList.qml
--- original/usr/share/lipstick-jolla-home-qt5/eventsview/EventsViewList.qml	2018-03-28 06:26:54.108045286 +0100
+++ patch/usr/share/lipstick-jolla-home-qt5/eventsview/EventsViewList.qml	2018-08-07 09:23:58.392122202 +0100
@@ -30,7 +30,7 @@ SilicaFlickable {
     property real defaultNotificationAreaY: headerColumn.y + headerColumn.height + Theme.paddingMedium
     property bool menuOpen: pullDownMenu != null && pullDownMenu.active
 
-    property bool _housekeepingAllowed: (notificationListModel.populated && notificationList.hasRemovableNotifications)
+    property bool _housekeepingAllowed: kmOn.value ? false :(notificationListModel.populated && notificationList.hasRemovableNotifications)
                         || feedsList.hasRemovableNotifications
 
     contentHeight: Math.ceil(Math.max(footerSpacer.y + footerSpacer.height,
@@ -175,7 +175,23 @@ SilicaFlickable {
             active: false
             eventsView: root.eventsView
         }
+    ConfigurationValue {
+        id: kmOn
+        key: "/desktop/lipstick-jolla-home/kidsMode/kmOn"
+        defaultValue: false
+        }
+        ConfigurationValue {
+        id: showEvNotifications
+
+        key: "/desktop/lipstick-jolla-home/kidsMode/"+currentUser.value+"/showEvNotifications"
+        defaultValue: false
+    }
+    ConfigurationValue {
+        id: currentUser
 
+        key: "/desktop/lipstick-jolla-home/kidsMode/currentUser"
+        defaultValue: 0
+        }
         ConfigurationValue {
             id: eventsScreenWidgets
 
@@ -231,7 +247,8 @@ SilicaFlickable {
 
     Item {
         id: notificationsArea
-
+        
+        visible: kmOn.value ? showEvNotifications.value: true
         width: root.notificationWidth
         height: Math.max(systemUpdateList.contentHeight + notificationList.contentHeight + feedsList.height, root.height - y)
 
diff -rupN original/usr/share/lipstick-jolla-home-qt5/eventsview/EventsView.qml patch/usr/share/lipstick-jolla-home-qt5/eventsview/EventsView.qml
--- original/usr/share/lipstick-jolla-home-qt5/eventsview/EventsView.qml	2018-04-09 10:15:50.244713634 +0100
+++ patch/usr/share/lipstick-jolla-home-qt5/eventsview/EventsView.qml	2018-04-09 10:20:05.519118591 +0100
@@ -213,6 +213,11 @@ MouseArea {
             defaultValue: true
         }
 
+        ConfigurationValue {
+            id: kmOn
+            key: "/desktop/lipstick-jolla-home/kidsMode/kmOn"
+            defaultValue: false
+        }
         Loader {
             id: shortcutLoader
             active: shortcutsOverlay.active
@@ -352,6 +357,7 @@ MouseArea {
                                 rightMargin: Theme.paddingLarge
                             }
                             icon.source: "image://theme/icon-s-setting"
+                            visible: !kmOn.value
 
                             onClicked: {
                                 favoriteSettingsLoader.showEventsSettings()
diff -rupN original/usr/share/lipstick-jolla-home-qt5/eventsview/FavoriteSettingsDelegate.qml patch/usr/share/lipstick-jolla-home-qt5/eventsview/FavoriteSettingsDelegate.qml
--- original/usr/share/lipstick-jolla-home-qt5/eventsview/FavoriteSettingsDelegate.qml	2018-04-10 06:27:32.177747256 +0100
+++ patch/usr/share/lipstick-jolla-home-qt5/eventsview/FavoriteSettingsDelegate.qml	2018-04-10 06:34:41.682729240 +0100
@@ -9,6 +9,7 @@ import QtQuick 2.0
 import Sailfish.Silica 1.0
 import com.jolla.settings 1.0
 import org.nemomobile.lipstick 0.1
+import org.nemomobile.configuration 1.0
 import "../main"
 
 EditableGridDelegate {
@@ -46,7 +47,12 @@ EditableGridDelegate {
             value: item.down && item.enabled
         }
     }
-
+    ConfigurationValue {
+        id: kmOn
+        key: "/desktop/lipstick-jolla-home/kidsMode/kmOn"
+        defaultValue: false
+    }
+    
     Image {
         id: removeButton
         property bool pressed
@@ -65,7 +71,7 @@ EditableGridDelegate {
             horizontalCenterOffset: removeButtonOffset.width
             verticalCenterOffset: removeButtonOffset.height
         }
-        enabled: Lipstick.compositor.eventsLayer.shortcutsHousekeeping
+        enabled: Lipstick.compositor.eventsLayer.shortcutsHousekeeping && !kmOn.value
         opacity: enabled ? 1.0 : 0
         Behavior on opacity { FadeAnimation { } }
     }
diff -rupN original/usr/share/lipstick-jolla-home-qt5/eventsview/QuickActionsLoader.qml patch/usr/share/lipstick-jolla-home-qt5/eventsview/QuickActionsLoader.qml
--- original/usr/share/lipstick-jolla-home-qt5/eventsview/QuickActionsLoader.qml	2018-04-10 06:27:32.137747830 +0100
+++ patch/usr/share/lipstick-jolla-home-qt5/eventsview/QuickActionsLoader.qml	2018-04-10 06:36:54.820457528 +0100
@@ -10,6 +10,7 @@ import Sailfish.Silica 1.0
 import org.nemomobile.lipstick 0.1
 import com.jolla.settings 1.0
 import com.jolla.settings.system 1.0
+import org.nemomobile.configuration 1.0
 import "../main"
 
 Loader {
@@ -25,7 +26,12 @@ Loader {
 
     asynchronous: true
     visible: status === Loader.Ready
-
+    ConfigurationValue {
+        id: kmOn
+        key: "/desktop/lipstick-jolla-home/kidsMode/kmOn"
+        defaultValue: false
+    }
+    
     Rectangle {
         anchors {
             fill: parent
@@ -148,7 +154,7 @@ Loader {
                             horizontalCenter: parent.right
                             horizontalCenterOffset: Theme.iconSizeMedium/2-iconOffset
                         }
-                        enabled: wrapper.editMode
+                        enabled: wrapper.editMode && !kmOn.value
                         opacity: enabled ? 1.0 : 0
                         icon.source: "image://theme/icon-m-clear"
                         width: icon.width
diff -rupN original/usr/share/lipstick-jolla-home-qt5/launcher/LauncherGrid.qml patch/usr/share/lipstick-jolla-home-qt5/launcher/LauncherGrid.qml
--- original/usr/share/lipstick-jolla-home-qt5/launcher/LauncherGrid.qml	2018-03-13 09:21:55.310802724 +0000
+++ patch/usr/share/lipstick-jolla-home-qt5/launcher/LauncherGrid.qml	2018-06-22 10:31:07.094756429 +0100
@@ -13,10 +13,15 @@ import Sailfish.Silica.private 1.0
 import Sailfish.Policy 1.0
 import Sailfish.Lipstick 1.0
 import "../main"
+import org.nemomobile.configuration 1.0
 
 SilicaGridView {
     id: gridview
-
+            ConfigurationGroup {
+                id: kidsModeSettings
+                path: "/desktop/lipstick-jolla-home/kidsMode"
+                property bool kmOn: false
+    }
     // The multipliers below for Large screens are magic. They look good on Jolla tablet.
     property real minimumCellWidth: Screen.sizeCategory >= Screen.Large ? Theme.itemSizeExtraLarge * 1.6 : Theme.itemSizeExtraLarge
     // phone reference row height: 960 / 6
@@ -225,7 +230,7 @@ SilicaGridView {
         }
 
         onPressAndHold: {
-            if (Lipstick.compositor.launcherLayer.active) {
+            if (Lipstick.compositor.launcherLayer.active  && !kmOn.value) {
                 setEditMode(true)
             }
         }
@@ -330,7 +335,7 @@ SilicaGridView {
                 anchors.horizontalCenter: parent.horizontalCenter
                 enabled: launcherEditMode && policy.value
                 opacity: enabled ? 1.0 : 0.0
-                visible: launcherEditMode && !isFolder && AppControl.isUninstallable(object.filePath) && !object.isUpdating
+                visible: launcherEditMode && !isFolder && AppControl.isUninstallable(object.filePath) && !object.isUpdating && !kidsModeSettings.kmOn 
                 onClicked: launcher.removeApplication(object.filePath, object.title)
             }
         }
diff -rupN original/usr/share/lipstick-jolla-home-qt5/launcher/Launcher.qml patch/usr/share/lipstick-jolla-home-qt5/launcher/Launcher.qml
--- original/usr/share/lipstick-jolla-home-qt5/launcher/Launcher.qml	2018-03-13 09:23:06.737044060 +0000
+++ patch/usr/share/lipstick-jolla-home-qt5/launcher/Launcher.qml	2018-06-22 10:31:19.516365486 +0100
@@ -13,10 +13,35 @@ import Sailfish.Policy 1.0
 import Sailfish.Lipstick 1.0
 import org.nemomobile.dbus 2.0
 import com.jolla.lipstick 0.1
+import org.nemomobile.configuration 1.0
 
 SilicaListView {
     id: launcherPager
-
+    ConfigurationValue {
+        id: kmOn
+        key: "/desktop/lipstick-jolla-home/kidsMode/kmOn"
+        defaultValue: false
+    }
+        
+    ConfigurationValue {
+        id: lockScreenShortcuts
+
+        key: "/desktop/lipstick-jolla-home/kidsMode/"+currentUser.value+"/appsList"
+        defaultValue: []
+    }
+    ConfigurationValue {
+        id: currentUser
+
+        key: "/desktop/lipstick-jolla-home/kidsMode/currentUser"
+        defaultValue: 0
+    }
+            
+    ConfigurationGroup {
+        id: kmTrigger
+path: "/desktop/lipstick-jolla-home/kidsMode"
+        property bool triggerLoad: false
+        onValueChanged:if(key == "triggerLoad") lFolder.load() 
+    }
     property bool launcherActive: Lipstick.compositor.launcherLayer.active
     onLauncherActiveChanged: if (!launcherActive) { resetPosition(400) }
 
@@ -112,7 +137,7 @@ SilicaListView {
         }
 
         onPressAndHold: {
-            if (Lipstick.compositor.launcherLayer.active &&
+            if (Lipstick.compositor.launcherLayer.active && !kmOn.value &&
                 Math.abs(mouse.x - pressX) < Theme.startDragDistance &&
                 Math.abs(mouse.y - pressY) < Theme.startDragDistance) {
                 launcher.setEditMode(true)
@@ -131,11 +156,14 @@ SilicaListView {
             gridManager.onStopScrolling: launcherPager.stopScrolling()
 
             model: LauncherFolderModel {
+                id: lFolder
                 iconDirectories: Theme.launcherIconDirectories
-                blacklistedApplications: {
+                blacklistedApplications:kmOn.value? getBlackList(lockScreenShortcuts.value):getBlackList([ ])
+
+                function getBlackList(appList) {
                     // Currently desktop-file path is good app grid item
                     // identifier. However, this is a subject to change in future.
-                    var blacklist = []
+                    var blacklist = appList 
                     var path = "/usr/share/applications"
 
                     if (!AccessPolicy.cameraEnabled) {
diff -rupN original/usr/share/lipstick-jolla-home-qt5/notifications/NotificationItem.qml patch/usr/share/lipstick-jolla-home-qt5/notifications/NotificationItem.qml
--- original/usr/share/lipstick-jolla-home-qt5/notifications/NotificationItem.qml	2018-07-23 11:31:58.999855635 +0100
+++ patch/usr/share/lipstick-jolla-home-qt5/notifications/NotificationItem.qml	2018-08-03 13:23:21.169151734 +0100
@@ -11,6 +11,7 @@ import Sailfish.Lipstick 1.0
 import org.nemomobile.dbus 2.0
 import com.jolla.lipstick 0.1
 import org.nemomobile.lipstick 0.1
+import org.nemomobile.configuration 1.0
 
 Item {
     id: root
@@ -42,6 +43,11 @@ Item {
     property int _expansionThreshold: 3
     property int _expansionMaximum: 15
 
+    ConfigurationValue {
+        id: kmOn
+        key: "/desktop/lipstick-jolla-home/kidsMode/kmOn"
+        defaultValue: false
+    }
     function _invokeAppAction() {
         // Invoke the 'app' action directly
         var remoteActions = root._notificationGroup.remoteActions
@@ -93,7 +99,7 @@ Item {
         }
 
         onTriggered: {
-            if (!root._notificationGroup || !root._notificationGroup.members.length) {
+            if (!root._notificationGroup || !root._notificationGroup.members.length || kmOn.value) {
                 return
             }
             root._invokeAppAction()
@@ -175,7 +181,7 @@ Item {
                     }
 
                     onTriggered: {
-                        notification.actionInvoked("default")
+                     if(!kmOn.value)   notification.actionInvoked("default")
                     }
                 }
             }
@@ -227,7 +233,7 @@ Item {
                 color: expansionToggle.highlighted ? Theme.secondaryHighlightColor : Theme.secondaryColor
             }
 
-            onClicked: root._invokeAppAction()
+            onClicked:if(!kmOn.value)  root._invokeAppAction()
         }
     }
 }
diff -rupN original/usr/share/lipstick-jolla-home-qt5/notifications/NotificationPreview.qml patch/usr/share/lipstick-jolla-home-qt5/notifications/NotificationPreview.qml
--- original/usr/share/lipstick-jolla-home-qt5/notifications/NotificationPreview.qml	2018-04-22 08:28:01.160162830 +0100
+++ patch/usr/share/lipstick-jolla-home-qt5/notifications/NotificationPreview.qml	2018-07-24 09:25:00.707667829 +0100
@@ -12,13 +12,14 @@ import com.jolla.lipstick 0.1
 import org.nemomobile.lipstick 0.1
 import org.nemomobile.thumbnailer 1.0
 import org.nemomobile.devicelock 1.0
+import org.nemomobile.configuration 1.0
 import "../systemwindow"
 
 SystemWindow {
     id: notificationWindow
 
     property QtObject notification: notificationPreviewPresenter.notification
-    property bool showNotification: notification != null && (notification.previewBody || notification.previewSummary)
+    property bool showNotification: notification != null && (notification.previewBody || notification.previewSummary) &&(kmOn.value? showPopUps.value:true)
     property string summaryText: showNotification ? notification.previewSummary : ''
     property string bodyText: showNotification ? notification.previewBody : ''
     property bool popupPresentation: state == "showPopup" || state == "hidePopup"
@@ -63,7 +64,23 @@ SystemWindow {
         property: "iconUrl"
         value: notificationWindow.iconUrl
     }
+    ConfigurationValue {
+        id: kmOn
+        key: "/desktop/lipstick-jolla-home/kidsMode/kmOn"
+        defaultValue: false
+    }
+ConfigurationValue {
+        id: showPopUps
 
+        key: "/desktop/lipstick-jolla-home/kidsMode/"+currentUser.value+"/showPopUps"
+        defaultValue: true
+    }
+    ConfigurationValue {
+        id: currentUser
+
+        key: "/desktop/lipstick-jolla-home/kidsMode/currentUser"
+        defaultValue: 0
+    }
     function firstLine(str) {
         var i = str.indexOf("\n")
         if (i >= 0) {
@@ -137,7 +154,7 @@ SystemWindow {
         }
 
         onClicked: {
-            if (notification) {
+            if (notification && !kmOn.value) {
                 notificationWindow._invoked = true
                 notification.actionInvoked("default")
 
diff -rupN original/usr/share/lipstick-jolla-home-qt5/switcher/Switcher.qml patch/usr/share/lipstick-jolla-home-qt5/switcher/Switcher.qml
--- original/usr/share/lipstick-jolla-home-qt5/switcher/Switcher.qml	2018-04-21 19:02:44.831496953 +0100
+++ patch/usr/share/lipstick-jolla-home-qt5/switcher/Switcher.qml	2018-06-19 12:53:38.330505873 +0100
@@ -12,6 +12,7 @@ import com.jolla.lipstick 0.1
 import Sailfish.Silica 1.0
 import Sailfish.Silica.private 1.0
 import Sailfish.Lipstick 1.0
+import org.nemomobile.configuration 1.0
 import "../compositor"
 import "../main"
 
@@ -177,11 +178,143 @@ SilicaFlickable {
             repeater.itemAt(lastIndex).close()
         }
     }
+    Timer {
+        id: closeNonKmTimer
+        property int lastIndex
+
+        interval: 100
+        repeat: true
+        triggeredOnStart: true
+
+        onTriggered: {
+            if (repeater.count === 0 || lastIndex === 0) {
+                running = false
+                return
+            }
 
+            lastIndex = lastIndex - 1
+            if (lastIndex > repeater.count - 1) {
+                lastIndex = repeater.count - 1
+            }
+            var window = Lipstick.compositor.windowForId(repeater.itemAt(lastIndex).windowId)
+            if(window == null) repeater.itemAt(lastIndex).close()
+            else if(appTitles.value.indexOf(window.title) != -1) repeater.itemAt(lastIndex).close()
+        }
+    }
+    property var switcherModelBackup: []
+    
+    function hideApps() {     
+        switcherRoot.minimizeLaunchingWindows()
+        switcherRoot.closePending = true 
+        if(kmOn.value) {
+            var i = 0       
+            var items =[]   
+            
+            while(i < repeater.count) {   
+                var item = repeater.itemAt(i)
+                var window = Lipstick.compositor.windowForId(item.windowId)
+                if(window == null) {
+                    i = i + 1
+                    continue
+                }
+              if(appTitles.value.indexOf(window.title) != -1)   {
+                    if(item.launcherItem == null) {
+                       // item.minimized = true
+                      //  item.close()
+                        i = i + 1
+                        continue
+                    }
+                   
+                    items.push(item.launcherItem)
+                    items.push(switcherModel.windowId(i))
+                    items.push(item.coverId)
+                    items.push(item.windowId)
+                    items.push(item.coverHint)
+                    items.push(item.processId)
+                    items.push(item.windowMapPending)
+                    items.push(item.showingWid)
+                    items.push(item.running)
+                    items.push(item.pending)
+                    items.push(item.minimized)
+                    items.push(item._persistCloseApplicationEnabled)
+                    switcherModelBackup.push(items)
+                    switcherModel.remove(i)
+                    items =[]          
+                    
+                }
+                else {
+                        i = i + 1
+                }
+            }
+                switcherRoot.closeAll()  
+        }
+        else {
+            for (var i = 0; i <switcherModelBackup.length; ++i) {
+                var j =repeater.count                
+                var item =switcherModelBackup[i][0]
+                switcherModel.append(item)
+                var repItem = repeater.itemAt(j)
+                repItem.coverId = switcherModelBackup[i][2]
+                repItem.windowId = switcherModelBackup[i][3]
+                repItem.coverHint = switcherModelBackup[i][4]
+                repItem.processId = switcherModelBackup[i][5]
+                repItem.windowMapPending = switcherModelBackup[i][6]
+                repItem.showingWid = switcherModelBackup[i][7]
+                repItem.running = switcherModelBackup[i][8]
+                repItem.pending = switcherModelBackup[i][9]
+                repItem.minimized = switcherModelBackup[i][10]
+                repItem._persistCloseApplicationEnabled = switcherModelBackup[i][11]
+            }
+            switcherModelBackup = []
+        }
+       //switcherRoot.closePending = false
+    }   
+
+    ConfigurationGroup {
+        id: kmTrigger
+path: "/desktop/lipstick-jolla-home/kidsMode"
+        property bool triggerClose: false
+        property bool closeAllApps: true
+        onValueChanged: if(key == "triggerClose") closeAllApps ? switcherRoot.closeAll() :switcherRoot.hideApps()
+            
+    }  
+    
+    ConfigurationValue {
+        id: kmOn
+        key: "/desktop/lipstick-jolla-home/kidsMode/kmOn"
+        defaultValue: false
+    }
+    
+    ConfigurationValue {
+        id: currentUser
+        key: "/desktop/lipstick-jolla-home/kidsMode/currentUser"
+        defaultValue: 0
+    }
+    
+    ConfigurationValue {
+        id: lockScreenShortcuts
+
+        key: "/desktop/lipstick-jolla-home/kidsMode/"+currentUser.value+"/appsList"
+        defaultValue: []
+    }
+    
+    ConfigurationValue {
+        id: appTitles
+
+        key: "/desktop/lipstick-jolla-home/kidsMode/"+currentUser.value+"/appTitles"
+        defaultValue: []
+    }
+    
     function closeAll() {
+        if(kmTrigger.closeAllApps) {
         closeAllTimer.stop()
         closeAllTimer.lastIndex = repeater.count
-        closeAllTimer.start()
+        closeAllTimer.start()}
+        else {
+         closeNonKmTimer.stop()
+        closeNonKmTimer.lastIndex = repeater.count
+        closeNonKmTimer.start()}
+        
     }
 
     function isAndroidApplication(launcherItem) {
